src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,293 @@
+import {Feature, Settings} from '../../utils/settings'
+import {Navigation} from '../../roam/navigation'
+import {browser} from 'webextension-polyfill-ts'
+import {createPopper, Instance} from '@popperjs/core'
+
+export const config: Feature = {
+    id: 'live_preview',
+    name: 'Live Preview',
+    warning: 'Experimental feature; Large databases might see performance issues.',
+    enabledByDefault: false,
+}
+const checkSettingsAndSetupIframeToggle = () => {
+    Settings.isActive('live_preview').then(active => {
+        toggleIframe(active)
+    })
+}
+checkSettingsAndSetupIframeToggle()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupIframeToggle()
+    }
+})
+
+let iframeInstance: PreviewIframe | null = null
+
+const toggleIframe = (active: boolean) => {
+    if (!iframeInstance) {
+        iframeInstance = new PreviewIframe()
+    }
+    if (active) {
+        iframeInstance.activate()
+    } else {
+        iframeInstance.destroy()
+        iframeInstance = null
+    }
+}
+
+class PreviewIframe {
+    iframeId = 'roam-toolkit-iframe-preview'
+    iframe: HTMLIFrameElement | null = null
```

Comment:
how about constructing this (and maybe calling activate in general?) as a part of class constructor, so we can can be sure that it's not null and avoid null-checking it everywhere?

Reply:
as after all, iframe should exist during the whole lifecycle of this class